pkg/infra/baseagent.go

```diff
@@ -59,7 +56,7 @@ func NewBaseAgent(world world.World, agentType int, agentHP int, agentFloor int,
 		floor:          agentFloor,
 		agentType:      agentType,
 		tower:          tower,
-		inbox:          list.New(),
+		inbox:          make(chan messages.Message, 15),
```

Comment:
Not sure about the best size to make this buffer - currently set to size 15. This is how many messages an inbox can store before it's full. Can make it larger?